spring-security/src/main/java/com/sap/cloud/security/spring/token/authentication/HybridJwtDecoder.java

```diff
@@ -82,7 +83,11 @@ public Jwt decode(String encodedToken) {
 		default -> throw new BadJwtException("Tokens issued by " + token.getService() + " service aren't supported.");
 		}
 		if (validationResult.isErroneous()) {
-			throw new BadJwtException("The token is invalid: " + validationResult.getErrorDescription());
+			if (validationResult.getErrorDescription().contains("Error retrieving token keys")) {
```

Comment:
We should probably check for [this](https://github.com/SAP/cloud-security-services-integration-library/blob/cf45ae45bb76e54b09aa315de7d3b07fbdcbbd4d/java-security/src/main/java/com/sap/cloud/security/token/validation/validators/JwtSignatureValidator.java#L48) error message instead because it is already doing a check for JWKS fetch failure there and it results in this higher level error message